File: src/com/adventofcode/december00/Input.java
package com.adventofcode.december00;

import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Input {

    private List<String> inputList;

    public Input(String fileName) {
        inputList = readFile("data/" + fileName);
    }

    public List<String> getInputList() {
        return inputList;
    }

    //-----------------------------------------------------------------------------------------------------------

    public BaseClass getBaseClass() {
        return new BaseClass(inputList);
    }


    //-----------------------------------------------------------------------------------------------------------

    private List<String> readFile(String fileName) {
        int lineCount = 0;
        List<String> inputList = new ArrayList<>();
        try {
            FileInputStream fis = new FileInputStream(fileName);
            BufferedReader br = new BufferedReader(new InputStreamReader(fis));
            String line;
            while ((line = br.readLine()) != null) {
                inputList.add(line);
            }
        } catch ( Exception e) {

        }
        return inputList;
    }
}

File: src/com/adventofcode/december00/Input.kt
package com.adventofcode.december00

import java.io.BufferedReader
import java.io.FileInputStream
import java.io.InputStreamReader

class Input(fileName: String) {

    private val inputList = readFile("data/$fileName")

    fun getInputList(): List<String> {
        return inputList
    }

    //-----------------------------------------------------------------------------------------------------------

    fun getBaseClass(): BaseClass {
        return BaseClass(inputList)
    }

    //-----------------------------------------------------------------------------------------------------------

    private fun readFile(fileName: String): List<String> {
        val lineCount = 0
        val inputList: MutableList<String> = ArrayList()
        try {
            val fis = FileInputStream(fileName)
            val br = BufferedReader(InputStreamReader(fis))
            var line: String
            while (br.readLine().also { line = it } != null) {
                inputList.add(line)
            }
        } catch (e: Exception) {
        }
        return inputList
    }
}